### PR TITLE
Remove includes in .cpp files for things the corresponding .h file already included

### PR DESCRIPTION
**Pull request by practicalswift**
Remove includes in .cpp files for things the corresponding .h file already included.

Example case:
* `addrdb.cpp` includes `addrdb.h` and `fs.h`
* `addrdb.h` includes `fs.h`

Then remove the direct inclusion of `fs.h` in `addrman.cpp` and rely on the indirect inclusion of `fs.h` via the included `addrdb.h`.

In line with the header include guideline (see #10575).

### CODE REVIEW

**Comment by practicalswift** [comment]
Candidates for removal found by:

```bash
#!/bin/bash

for H in $(git grep "" -- "*.h" | cut -f1 -d: | uniq); do
    CPP=${H/%\.h/.cpp}
    if [[ ! -e $CPP ]]; then
        continue
    fi
    cat "$H" "$CPP" | grep -E "^#include " | sort | uniq -c | grep " 2 #include" && echo "$CPP" && echo
done
```

**Comment by practicalswift** [comment]
Whoops, force pushed to the wrong branch. Correction coming up!

**Comment by practicalswift** [comment]
Rebased!

**Comment by practicalswift** [comment]
Updated!

**Comment by practicalswift** [comment]
Anyone willing to review? :-)

**Comment by razamobin** [comment]
Hi, I'm new to the bitcoin source code, but I looked for redundant includes and got the same changeset as you. Also I was able to run unit tests and they all passed (although I guess Travis did this already?). Either way, looks good to me.

**Comment by luke-jr** [comment]
Soft NACK. IMO it's better code quality to explicitly include everything needed in each source file using them.

**Comment by practicalswift** [comment]
Rebased! :-)

**Comment by danra** [comment]
I agree with luke-jr

**Comment by sipa** [comment]
@luke-jr @danra Then you should argue to change the guidelines in developer-notes.md:

> Every .cpp and .h file should #include every header file it directly uses classes, functions or other definitions from, even if those headers are already included indirectly through other headers. One exception is that a .cpp file does not need to re-include the includes already included in its corresponding .h file.

I personally think that rule makes sense, as you can see the .cpp file as the continuation of the corresponding .h file.

**Comment by danra** [comment]
@sipa Yes, I'm for changing this rule as it creates an unnecessary dependency of the implementation (.cpp) on the interface (.h). If the interface no longer uses some other include directly, but the implementation still uses that include directly, removing the include from the interface would require adding it to the implementation.

(Note I'm not for having the implementation re-include *every* include of the interface - only those which it uses directly)

**Comment by sipa** [comment]
@danra I understand that point of view, but I think it's a non-issue. When the interface changes, almost certainly the implementation needs to change anyway.

**Comment by practicalswift** [comment]
Rebased! :-)

**Comment by practicalswift** [comment]
Rebased! Getting this merged would get rid of 76 lines :-)

**Comment by laanwj** [comment]
utACK a720b928c80f18d340173f39f63e7ef9cfb367c1

File: src/qt/platformstyle.cpp [hunk 1]
```diff
@@ -8,10 +8,8 @@
 
 #include <QApplication>
 #include <QColor>
-#include <QIcon>
 #include <QImage>
 #include <QPalette>
-#include <QPixmap>
```
Comment (promag) [inline src/qt/platformstyle.cpp]: This is in the header BUT I believe it's not necessary there. IMO the first exercise should be to reduce to a minimal the includes in a header file.

**Pull request merged**